core/src/main/java/org/apache/iceberg/ManifestFiles.java

```diff
@@ -141,7 +141,23 @@ public static ManifestReader<DataFile> read(
    * @return a manifest writer
    */
   public static ManifestWriter<DataFile> write(PartitionSpec spec, OutputFile outputFile) {
```

Comment:
Question to the reviewers: should we deprecate this method?

Reply:
@rdblue should we deprecate the old methods that do not accept `compressionCodec` and `compressionLevel`?

Reply:
No need for deprecation as per https://github.com/apache/iceberg/pull/5893#discussion_r989550192